dan_layer/storage_sqlite/src/sqlite_global_db_backend_adapter.rs

```diff
@@ -105,6 +106,26 @@ impl GlobalDbBackendAdapter for SqliteGlobalDbBackendAdapter {
         Ok(row.map(|r| r.value))
     }
 
+    fn get_data_with_connection(
+        &self,
+        key: &[u8],
+        connection: &SqliteConnection,
```

Comment:
It's more useful to provide a transaction so that multiple calls use it before committing. Also, there is no method for getting a connection, but there is a method for getting a transaction. And finally, the connection can be retrieved from the transaction

```suggestion
        tx: &Self::BackendTransaction,
```